core-bundle/tests/Doctrine/DoctrineTestCase.php

```diff
@@ -12,256 +12,61 @@
 use Doctrine\DBAL\Connection;
 use Doctrine\DBAL\Platforms\MySqlPlatform;
-use Doctrine\DBAL\Schema\AbstractSchemaManager;
-use Doctrine\DBAL\Schema\MySqlSchemaManager;
-use Doctrine\DBAL\Statement;
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\DBAL\Schema\SchemaConfig;
 use Doctrine\ORM\Configuration;
-use Doctrine\ORM\EntityManagerInterface;
-use Doctrine\ORM\Mapping\ClassMetadataFactory;
-use Doctrine\ORM\Mapping\DefaultQuoteStrategy;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\Mapping\Driver\AnnotationDriver;
+use Doctrine\Persistence\Mapping\Driver\MappingDriverChain;
 use PHPUnit\Framework\MockObject\MockObject;
 
 abstract class DoctrineTestCase extends TestCase
 {
     /**
      * Mocks a Doctrine registry with database connection.
      *
-     * @return Registry&MockObject
-     */
-    protected function mockDoctrineRegistry(Statement $statement = null, string $filter = null): Registry
-    {
-        $schemaManager = $this->createMock(AbstractSchemaManager::class);
-        $schemaManager
-            ->method('tablesExist')
-            ->willReturn(true)
-        ;
-
-        $config = $this->createMock(Configuration::class);
-        $config
-            ->method('getSchemaAssetsFilter')
-            ->willReturn($this->getSchemaAssetsFilter($filter))
-        ;
-
-        $connection = $this->createMock(Connection::class);
-        $connection
-            ->method('getDatabasePlatform')
-            ->willReturn(new MySqlPlatform())
-        ;
-
-        $connection
-            ->method('executeQuery')
-            ->willReturn($statement)
-        ;
-
-        $connection
-            ->method('getSchemaManager')
-            ->willReturn($schemaManager)
-        ;
-
-        $connection
-            ->method('getParams')
-            ->willReturn(
-                [
-                    'defaultTableOptions' => [
-                        'charset' => 'utf8mb4',
-                        'collate' => 'utf8mb4_unicode_ci',
-                    ],
-                ]
-            )
-        ;
-
-        $connection
-            ->method('getConfiguration')
-            ->willReturn($config)
-        ;
-
-        $registry = $this->createMock(Registry::class);
-        $registry
-            ->method('getConnection')
-            ->willReturn($connection)
-        ;
-
-        $registry
-            ->method('getConnections')
-            ->willReturn([$connection])
-        ;
-
-        $registry
-            ->method('getManagerNames')
-            ->willReturn([])
-        ;
-
-        return $registry;
-    }
-
-    /**
-     * Mocks a Doctrine registry with database connection and ORM.
+     * @param Connection|MockObject|null $connection
      *
      * @return Registry&MockObject
      */
-    protected function mockDoctrineRegistryWithOrm(array $metadata = [], string $filter = null): Registry
+    protected function mockDoctrineRegistry($connection = null): Registry
```

Comment:
Changing the method signature is a BC break. Not sure if our BC promise applies to test class, but since it is an abstract class, we must assume that people are using it. @contao/developers WDYT?

Reply:
Imho tests should never be part of the BC promise because they mostly/only make sense in context of the tested code. Otherwise you would end up introducing abstractions and BC layers that shouldn't exist in test code.

Symfony exludes them as well (see [docs](https://symfony.com/doc/current/contributing/code/bc.html#using-our-classes)):
> Classes, properties and methods that bear the tag @internal as well as the classes located in the various *\Tests\ namespaces are an exception to this rule. They are meant for internal use only and should not be accessed by your own code.

Reply:
See https://github.com/contao/docs/pull/648